### DISCUSSION TITLE
Update app state via javascript

### DISCUSSION THREAD

**Discussion by diedder** [Budibase/budibase#6475] [Help] [answered]
Hi,

is it possible to update the app state via javascript? I want to set the state when clicking a link to have the last url for jumping back to the last site. With buttons it works fine but with using a link I have only the chance to do this via javascript, but a normal variable assignment fails...

thanks for help!

**Comment by melohagan** [comment, marked as answer]
Hi @diedder 

As far as I'm aware there's no way to do that. If you use a button instead of a link, then you could use the **Update State** action and the **Navigate To** action.
You'd need to style the button to look like a link.

There's an issue relating to this here: https://github.com/Budibase/budibase/issues/5420

**Reply by diedder** [reply]
Thank you for your quick answer an the button tip @melohagan.
My CSS skills are very poor, but I'll give it a try ;)
